ui/perfherder/CompareSubtestDistributionView.jsx

```diff
@@ -0,0 +1,304 @@
+          data: [],
+          title: `${targetId} Replicates`,
+          chart_type: 'missing-data',
+          missing_text: 'No Data Found',
+          width: 1000,
+          height: 275,
+        };
+
+  fetchReplicateGraphData = async (project, revision, subtestSignature) => {
+    const replicateData = {};
+
+    const response = await PushModel.getList({ repo: project, revision });
+    const { results: pushPatch } = await response.json();
+    const [first] = pushPatch;
+    replicateData.resultSet = first;
+
+    const perfDatumList = await PerfSeriesModel.getSeriesData(project, {
+      signatures: subtestSignature,
+      push_id: replicateData.resultSet.id,
+    });
+
+    if (!perfDatumList[subtestSignature]) {
+      replicateData.replicateDataError = true;
+      return;
+    }
+    const numRuns = perfDatumList[subtestSignature].length;
+    const replicatePromises = perfDatumList[subtestSignature].map(value =>
+      PerfSeriesModel.getReplicateData({ job_id: value.job_id }),
+    );
+
+    return Promise.all(replicatePromises).then(
+      localReplicateData => {
+        let replicateValues = localReplicateData.concat.apply(
+          [],
+          localReplicateData.map(data => {
+            const testSuite = data.suites.find(
+              suite => suite.name === this.state.testSuite,
+            );
+            const subtest = testSuite.subtests.find(
+              subtest => subtest.name === this.state.subtest,
+            );
+            return subtest.replicates;
+          }),
+        );
+        // metrics-graphics doesn't accept "0" as x_accesor
+        replicateValues = replicateValues.map((value, index) => ({
+          replicate: (index + 1).toString(),
+          value,
+        }));
+
+        return {
+          replicateData,
+          drawingData: { numRuns, replicateValues },
+        };
+      },
+      () => {
+        replicateData.replicateDataError = true;
+      },
+    );
+  };
+
+  render() {
+    const {
+      originalProject,
+      newProject,
+      originalResultSet,
+      newResultSet,
+      originalReplicateError,
+      newReplicateError,
+      dataLoading,
+      platform,
+      testName,
+      graphOneSpecs,
+      graphTwoSpecs,
+      baseDrawingData,
+      newDrawingData,
+    } = this.state;
+    const { originalRevision, newRevision } = this.props.$stateParams;
+    const baseData =
+      baseDrawingData && baseDrawingData.replicateValues
+        ? baseDrawingData.replicateValues
+        : [];
+    const newData =
+      newDrawingData && newDrawingData.replicateValues
+        ? newDrawingData.replicateValues
+        : [];
+    // debugger; // eslint-disable-line no-debugger
+    return (
+      originalRevision &&
+      newRevision && (
+        <div className="container-fluid">
+          {dataLoading ? (
+            <div>
+              Loading all results, please wait a minute...
+              <img alt="dancing cat" src={dancing_cat} />
```

Comment:
I think we should replace the dancing cat with something consistent with TH. Either the spinner used in Intermittent Failures View or you can use the new Bootstrap spinner class.